app/src/main/java/com/simply/birthdayapp/commondomain/local/DataStoreProvider.kt

```diff
@@ -5,4 +5,6 @@ import kotlinx.coroutines.flow.Flow
 interface DataStoreProvider {
     fun isSignedIn(): Flow<Boolean>
     suspend fun setSignedIn(isSignedIn: Boolean)
+    suspend fun saveAccessToken(token: String)
+    suspend fun getToken() : Flow<String>
```

Comment:
it is not a good practice to have suspend function and return flow, for your code you can remove suspend, and your function will be like this

 override fun getToken(): Flow<String> = dataStore.data.map { preferences ->
        preferences[ACCESS_TOKEN_KEY] ?: ""
    }.flowOn(Dispatchers.IO)